### PR TITLE
Reduce tracing spans to the debug level

### PR DESCRIPTION
**Pull request by olix0r**
We currently set almost all span metadata at the info level, meaning
that these spans are recorded/maintained for all connections/requests
even though there are virtually no logs emitted with this metadata at
the info level.

This changes all of these spans to only be recorded at the debug level.
The one exception is that the per-connection `accept` span is maintained
at the info level so that errors are reported with at least
connection-level metadata.

### CODE REVIEW

File: linkerd/app/outbound/src/tcp/tests.rs [hunk 1]
```diff
@@ -296,7 +296,7 @@ async fn load_balances() {
         .map(|i| {
             tokio::spawn(
                 hello_world_client(svc_addr, &mut server)
-                    .instrument(tracing::info_span!("conn", i)),
+                    .instrument(tracing::debug_span!("conn", i)),
```
Comment (hawkw) [inline linkerd/app/outbound/src/tcp/tests.rs:299]: IMO it's nice to have the spans in these tests on by default, was the change to the tests intended, or was it just an accident of find-and-replace?

File: linkerd/app/src/lib.rs [hunk 2]
```diff
@@ -86,31 +86,32 @@ impl Config {
...
-            info_span!("opencensus")
+            debug_span!("opencensus")
                 .in_scope(|| oc_collector.build(identity, dns, metrics, client_metrics))
         }?;
 
         let admin = {
             let identity = identity.local();
             let drain = drain_rx.clone();
-            info_span!("admin").in_scope(move || admin.build(identity, report, log_level, drain))?
+            debug_span!("admin")
```
Comment (hawkw) [inline linkerd/app/src/lib.rs:119]: seems potentially worth keeping these top-level spans at INFO? i think it's mainly the ones in the actual data path where there's a noticeable latency impact?

File: linkerd/app/src/lib.rs [hunk 3]
```diff
@@ -144,7 +145,7 @@ impl Config {
                 outbound_metrics.clone(),
             );
 
-            let span = info_span!("outbound");
+            let span = debug_span!("outbound");
```
Comment (hawkw) [inline linkerd/app/src/lib.rs:154]: similarly, I think it's worth keeping the outbound and inbound spans at INFO for errors/warnings, as well as the accept span?

File: linkerd/app/test/src/connect.rs [hunk 4]
```diff
@@ -32,7 +32,7 @@ where
 
     fn call(&mut self, endpoint: E) -> Self::Future {
         let addr = endpoint.clone().into();
-        let span = tracing::info_span!("connect", %addr);
+        let span = tracing::debug_span!("connect", %addr);
```
Comment (hawkw) [inline linkerd/app/test/src/connect.rs:35]: this is a test, and i think it's useful to have these traces in the test on by default

File: linkerd/service-profiles/src/client.rs [hunk 5]
```diff
@@ -256,7 +256,7 @@ where
         mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
     ) -> Poll<Result<Profile, Error>> {
-        let span = info_span!("poll_profile");
+        let span = debug_span!("poll_profile");
```
Comment (hawkw) [inline linkerd/service-profiles/src/client.rs:259]: this one honestly seems like it should probably have been at trace from day one, that seems pretty verbose...

File: linkerd/app/gateway/src/config.rs [hunk 6]
```diff
@@ -51,7 +51,7 @@ impl Config {
                 Allow(self.allow_discovery),
             ))
             .check_new_service::<inbound::Target, http::Request<http::boxed::Payload>>()
-            .instrument(|_: &inbound::Target| info_span!("gateway"))
+            .instrument(|_: &inbound::Target| debug_span!("gateway"))
```
Comment (hawkw) [inline linkerd/app/gateway/src/config.rs:54]: IMO we might want to keep this one on at info, so we can tell what errors/warnings come from the gateway...

**Review by hawkw — COMMENTED** [review]
for the most part, i'm onboard with making the request-path spans disabled with the default filters. however, i noticed a few changes that i'm not convinced about:
- "top-level" spans (e.g. inbound, outbound, gateway, admin tasks) are now debug. i think it's probably worth keeping these at info for errors/warnings, and having just those on by default (+ accept) is probably not a huge impact compared to having a bunch of spans at every stack layer.
- a lot of spans in test code were also demoted to debug. i think it's nice to have these on by default when tests fail, but that's not a huge blocker.

also, might be worth rdoing a perf comparison again, since #726 should have also decreased the latency impact from tracing a bit...

File: linkerd/app/gateway/src/config.rs (hunk 6 above)
Comment (olix0r) [reply linkerd/app/gateway/src/config.rs:54]: I don't this is really practically a concern... You're either looking at a gateway pod or not.

File: linkerd/app/outbound/src/tcp/tests.rs (hunk 1 above)
Comment (olix0r) [reply linkerd/app/outbound/src/tcp/tests.rs]: yeah, global replace. restored test spans.

**Review by hawkw — APPROVED** [review]
⛵

**Pull request merged**